js/start.js

```diff
@@ -362,32 +378,40 @@ function setPublishingStatus(msg) {
 
 app.apiSocket.on('message', (e) => {
   if (e.jsonData) {
-    if (e.jsonData.status === 'publishing') {
-      setPublishingStatus({
-        msg: 'Publishing...',
-        type: 'message',
-      });
-
-      unpublishedContent = true;
-    } else if (e.jsonData.status === 'error publishing') {
-      setPublishingStatus({
-        msg: 'Publishing failed. <a class="js-retry">Retry</a>',
-        type: 'warning',
-      });
-
-      unpublishedContent = true;
-    } else if (e.jsonData.status === 'publish complete') {
-      setPublishingStatus({
-        msg: 'Publishing complete.',
-        type: 'message',
-      });
-
-      unpublishedContent = false;
-
-      publishingStatusMsgRemoveTimer = setTimeout(() => {
-        publishingStatusMsg.remove();
-        publishingStatusMsg = null;
-      }, 2000);
+    if (e.jsonData.status) {
+      if (e.jsonData.status === 'publishing') {
+        setPublishingStatus({
+          msg: 'Publishing...',
+          type: 'message',
+        });
+
+        unpublishedContent = true;
+      } else if (e.jsonData.status === 'error publishing') {
+        setPublishingStatus({
+          msg: 'Publishing failed. <a class="js-retry">Retry</a>',
+          type: 'warning',
+        });
+
+        unpublishedContent = true;
+      } else if (e.jsonData.status === 'publish complete') {
+        setPublishingStatus({
+          msg: 'Publishing complete.',
+          type: 'message',
+        });
+
+        unpublishedContent = false;
+
+        publishingStatusMsgRemoveTimer = setTimeout(() => {
+          publishingStatusMsg.remove();
+          publishingStatusMsg = null;
+        }, 2000);
+      }
+    } else if (e.jsonData.notification) {
```

Comment:
These other socket events are publish specific and this block is part of the `// manage publishing sockets` section. Can we move the follow block into it's own message handler (`app.apiSocket.on('message', (e) => {`...) and put it next to the other follow functionality (e.g. new function you create above)? I think the idea is to keep specific bits of functionality together, otherwise things get confusing if we interweave different blocks together.

I am hoping to break up start.js a bit into some sub modules and keeping the functionality as sand boxed as possible would be helpful when I get to that.